dfusion_rust_core/src/models/account_state.rs

```diff
@@ -92,14 +118,30 @@ impl AccountState {
         self.state_index = self.state_index.saturating_add(U256::one());
         self.state_hash = self.rolling_hash(self.state_index.low_u32());
     }
+
+    fn modify_balance<F>(&mut self, account_id: u16, token_id: u8, func: F)
+    where
+        F: FnOnce(&mut u128),
+    {
+        match self.balances.entry(U256::from(account_id)) {
+            Entry::Occupied(mut account) => match account.get_mut().entry(token_id) {
+                Entry::Occupied(mut balance) => func(balance.get_mut()),
+                Entry::Vacant(_) => panic!(
```

Comment:
Why aren't vacant balances just to be assumed 0 and we apply the function on 0?

Reply:
Because at the moment we don't have to use this assumption and so I wanted the logic to be rather strict (have the behavior as tight as possible and as lose as necessary). If there is a good use case in the future where we insert into 0 balances that don't yet explicitly exist I think we can change this behaviour.